ui/app/routes/vault/cluster/access/oidc.js

```diff
@@ -1,28 +1,3 @@
 import Route from '@ember/routing/route';
-import { inject as service } from '@ember/service';
 
-export default class OidcConfigureRoute extends Route {
-  @service router;
-
-  header = null;
-
-  beforeModel(transition) {
-    // set correct header state based on child route
-    // when no clients have been created display button as call to action to create
-    // list views share the same header with tabs as resource links
-    // the remaining routes are responsible for their own header
-    const routeName = transition.to.name;
-    if (routeName.includes('oidc.index')) {
-      this.header = 'cta';
-    } else {
-      const isList = ['clients', 'assignments', 'keys', 'scopes', 'providers'].find((resource) => {
-        return routeName.includes(`${resource}.index`);
-      });
-      this.header = isList ? 'list' : null;
-    }
-  }
-
-  setupController(controller) {
-    controller.header = this.header;
-  }
```

Comment:
I discovered while testing scope creation that the route hooks were not being triggered when transitioning from the scope create view to the list view and then going back to the create view with the browser back button. This was causing the header for the list view to show incorrectly on the create route. To fix this I moved the logic to the oidc controller and used the router service `routeDidChange` event instead.